src/main/java/io/confluent/connect/jdbc/JdbcUtils.java

```diff
@@ -193,5 +196,41 @@ public static String getIdentifierQuoteString(Connection connection) throws SQLE
   public static String quoteString(String orig, String quote) {
     return quote + orig + quote;
   }
+
+  /**
+   * Return current time at the database
+   * @param conn
+   * @param cal
+   * @return
+   */
+  public static Timestamp getCurrentTimeOnDB(Connection conn, Calendar cal) throws SQLException, ConnectException {
+
+    Statement stmt = conn.createStatement();
+    ResultSet rs;
+    String query;
+
+    // This is ugly, but to run a function, everyone does 'select function()'
+    // except Oracle that does 'select function() from dual'
+    // and Derby uses either the dummy table SYSIBM.SYSDUMMY1  or values expression (I chose to use values)
+    String dbProduct = conn.getMetaData().getDatabaseProductName();
+    if ("Oracle".equals(dbProduct))
+      query = "select CURRENT_TIMESTAMP from dual";
+    else if ("Apache Derby".equals(dbProduct))
+      query = "values(CURRENT_TIMESTAMP)";
+    else
+      query = "select CURRENT_TIMESTAMP ";
+
+    try {
+        rs = stmt.executeQuery(query);
+        if (rs.next())
+          return rs.getTimestamp(1, cal);
+        else
+          throw new ConnectException("Unable to get current time from DB using query " + query + " on database " + dbProduct);
+    } catch (SQLException e) {
+      log.error("Failed to get current time from DB using query " + query + " on database " + dbProduct, e);
+      throw e;
```

Comment:
So, we are definitely propagating SQLExceptions from executeQuery (where this is called) all the way up to poll() where we log them as "Failed to run query"...

If we want to handle this failed query differently than others, than having a separate exception makes sense, otherwise, it may get a bit confusing (Why do some query failures throw different exception than others?)